include/ensmallen_bits/callbacks/progress_bar.hpp

```diff
@@ -164,8 +164,22 @@ class ProgressBar
     output << "] " << progress << "% - ETA: " << (size_t) stepTimer.toc() *
         (epochSize - step + 1) % 60 << "s - loss: " <<
         objective / (double) step <<  "\r";
-    output.flush();
 
+    if (optimizer.MaxIterations() < function.NumFunctions() &&
+        optimizer.MaxIterations() != 0)
+    {
+      if (progress == (size_t)((double)(optimizer.MaxIterations()) /
+                                (double)(function.NumFunctions()) * 100) &&
+                      (int)(steps * optimizer.BatchSize() -
```

Comment:
Hi, @shrit I made a minor change in this line changing `step` to `steps` and got the expected output i.e.
```
Epoch 1/6
157/1875 [========>...........................................................................................] 8% - ETA: 0s - loss: 2039.62
Optimization terminated because of the entire dataset not being passed to the optimizer.
```
and now also if `ENS_PRINT_INFO` is turned on then the output is,
```
157/1875 [========>...........................................................................................] 8% - ETA: 0s - loss: 2039.62
Optimization terminated because of the entire dataset not being passed to the optimizer.
SGD: maximum iterations (10000) reached; terminating optimization.
```
Let me know what you think of this approach. Thanks.